src/util/cuda_vector_matrix.cu

```diff
@@ -16,10 +17,13 @@ namespace micm
     }
     int CopyToDevice(CudaVectorMatrixParam& param, std::vector<double>& h_data)
     {
-      return cudaMemcpy(param.d_data_, h_data.data(), sizeof(double) * param.num_elements_, cudaMemcpyHostToDevice);
+      cudaError_t err = cudaMemcpy(param.d_data_, h_data.data(), sizeof(double) * param.num_elements_, cudaMemcpyHostToDevice);
+      cudaDeviceSynchronize();
+      return err;
     }
     int CopyToHost(CudaVectorMatrixParam& param, std::vector<double>& h_data)
     {
+      cudaDeviceSynchronize();
```

Comment:
Is there a need for explicit device synchronization on memory copies?  I'm not seeing asynchronous behavior so this may be an issue of PEBKAC on my part.  😄

Reply:
You are correct that there is no need to do device synchronization for `CopyToDevice` so far unless we do asynchronous copy. I thought we might need it in the future and just added it here. But we may never need it and it seems to confuse people here. Just remove it for clarity.